### PR TITLE
feat: greet user with current time

### PR DESCRIPTION
**Pull request by TejasShekar**
1. Added context for user
2. Added greeting message based on time

Ignore the weather text. weather feature coming soon.

### CODE REVIEW

**Review by xii-vi — APPROVED** [review]
Great folder structure.
Greetings.js is thoughtful addition.
You can merge.

File: src/App.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 
 function App() {
-  const [user, setUser] = useState(null);
-
+  const { user, setUser } = useUserInfo();
   useEffect(() => {
     const user = localStorage.getItem("p8extension");
     setUser(user);
-  }, []);
+  });
```
Comment (darshandamre) [inline src/App.js:12]: do this in user context itself and you don't set the user in localstorage

File: src/utils/greetings.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    case 23:
+    case 24:
+      return "Good Night";
+    case 0:
+    case 1:
+    case 2:
+    case 3:
+    case 4:
+    default:
+      return "Hello";
```
Comment (darshandamre) [inline src/utils/greetings.js:33]:
```js
switch (true) {
    case 5 <= hour && hour <= 11:
      return "Good Morning";
    case 12 <= hour && hour <= 15:
      return "Good Afternoon";
    case 16 <= hour && hour <= 20:
      return "Good Evening";
    case 21 <= hour && hour <= 24:
      return "Good Night";
    default:
      return "Hello";
  }
```
better to write this switch like this

Comment (TejasShekar) [reply src/utils/greetings.js:33]: This looks better. Did not know we can write switch this way as well. Thankyou.

**Pull request merged**